configs/testdata/invalid-files/provider-localname-normalization.tf

```diff
@@ -0,0 +1,20 @@
+terraform {
+  required_providers {
+    test = {
```

Comment:
Should we have another similar test where the local name in `required_providers` is non-normalized?

```hcl
  required_providers {
    TEST = { ... }
  }
```

I can see from the other test fixture updates you made here that this _is_ now being caught, but it doesn't look like we have an expected-failure test for this particular part. (I'm sorry if I've just missed it...)

Reply:
Good call, absolutely, I'll add that!

Reply:
Adding that test confirmed something I had _thought_ I saw yesterday, then forgot about: `decodeRequiredProviders` was a bit inconsistent about generating an FQN for a provider when it encountered an error (and didn't have a source), so I made a small adjustment to make it more consistent. It doesn't _matter_, because we're going to stop and display the errors anyway, but it pleased me to fix anyway.